packages/react/src/drafts/IssueLabel/getColorFromHex.ts

```diff
@@ -0,0 +1,137 @@
+import {getContrast} from 'color2k'
+import {Hsluv} from 'hsluv'
+
+/**
+ * transforms a hex color provided by the user into a color object with background and text colors
+ * @param colorHex — the hex color provided by the user
+ * @param colorScheme — the color scheme the user has currently set
+ * @param isSelected
+ * @param bgColor — the background color from the selected theme, needed to calc the contrast for the colors
+ * @returns
+ */
+export const getColorsFromHex = (
+  colorHex: `#{string}`,
+  colorScheme: string = 'light',
+  bgColor: string,
+): React.CSSProperties | undefined => {
+  // start values for light mode
+  let bgLightness = 96
+  let lightnessIncrement: 1 | -1 = -1
+  let ratio = 4.5
+  // start values for dark mode
+  if (colorScheme.startsWith('dark')) {
+    bgLightness = 16
+    lightnessIncrement = 1
+    ratio = 5.5
+  }
+
+  // get hue and saturation value from hex color
+  // eslint-disable-next-line prefer-const
+  let {h, s} = hexToHsluv(colorHex)
+
+  // avoid intense bright colors
+  if (h >= 58 && h <= 198 && s > 70) {
+    s = 70
+  }
+  /**
+   * creating a background color from the provided hex color
+   */
+  const {colorHex: backgroundColor, lightness: currentBgLightness} = getColorWithContrast(
+    hsluvToHex({h, s, l: bgLightness}),
+    bgColor,
+    1.2,
+    lightnessIncrement as 1 | -1,
```

Comment:
Is this needed anymore now that the `lightnessIncrement` has the type above? 👀 

```suggestion
    lightnessIncrement,
```